web-ui/src/main/resources/catalog/components/search/map/MapFieldDirective.js

```diff
@@ -21,174 +21,179 @@
+               */
+              scope.$watch(scope.gnDrawBboxBtn, function (v) {
+                if (!v) {
+                  delete scope.searchObj.params.relation;
                 }
+                scope.currentExtent = scope.$eval(scope.gnDrawBboxBtn);
               });
 
-              // When search form is reset, remove the geom
-              scope.$on('beforeSearchReset', function(event, preserveGeometrySearch) {
-                if (!preserveGeometrySearch) {
-                  resetSpatialFilter();
-                  scope.interaction.active = false;
+              /**
+               * Set active relation (intersect, within, etc..). Run search
+               * when changed.
+               */
+              scope.setRelation = function (rel) {
+                scope.searchObj.params.relation = rel;
+                if (!!scope.searchObj.params.geometry) {
+                  scope.triggerSearch();
                 }
-              });
+              };
             }
           };
         }
-      ]);
+      };
+    }
+  ])
+
+    .directive('gnDrawBboxBtn', [
+      'olDecorateInteraction',
+      '$parse',
+      '$translate',
+      'gnSearchSettings',
+      'gnMapFieldService',
+      'gnMap',
+      function (olDecorateInteraction, $parse, $translate, gnSearchSettings, gnMapFieldService) {
+        return {
+          restrict: 'A',
+          scope: true,
+          controller: ['$scope', function ($scope) {
+            var dragbox = new ol.interaction.DragBox({
+              style: gnSearchSettings.olStyles.drawBbox
+            });
+            olDecorateInteraction(dragbox, $scope.map);
+            dragbox.active = false;
+            $scope.map.addInteraction(dragbox);
+            $scope.interaction = dragbox;
+          }],
+          link: function (scope, element, attrs) {
+
+            var parent = scope.$parent.$parent;
+
+            // Assign drawn extent to given scope property
+            var bboxGet = $parse(attrs['gnDrawBboxBtn']);
+            var bboxSet = bboxGet.assign;
+
+            // Create overlay to persist the bbox
+            var feature = new ol.Feature();
+            var featureOverlay = new ol.layer.Vector({
+              source: new ol.source.Vector(),
+              map: scope.map,
+              style: gnSearchSettings.olStyles.drawBbox
+            });
+            featureOverlay.getSource().addFeature(feature);
+
+            /**
+             * Update extent scope value with the WKT polygon
+             * @param {ol.geometry} geom
+             */
+            var updateField = function (geom) {
+              feature.setGeometry(geom);
+
+              // Get projection code of current map view
+              var proj = scope.map.getView().getProjection().getCode();
+
+              // Transform extent to WGS 1984 (EPSG:4326) if needed
+              var lonlatFeat, writer, wkt;
+              lonlatFeat = gnMapFieldService.toLonLat(feature, proj);
```

Comment:
Along with the service dependency injection, this really is the only piece of code I changed.
I did apply document formatting however, which removed redundant indents.